restler-core/src/main/java/net/researchgate/restdsl/validation/RestlerPreconditions.java

```diff
@@ -8,6 +8,13 @@
  * Restler-specific precondtions
  */
 public class RestlerPreconditions {
+
+    public static void ensureArgument(boolean value, String message) throws RestDslException {
+        if (!value) {
+            throw new RestDslException("Precondition failed: " + message, RestDslException.Type.ENTITY_ERROR);
```

Comment:
I don't think ENTITY_ERROR is correct here. Maybe PARAMS_ERROR would be better?

Reply:
Hm, the intent here is to be able to concisely validate an entity that gets POSTed to the service. 
Similarly to the other methods, where we check fields separately for existence or nonExistence of fields. 
This will just allow to be more flexible about the requirements, like date.after(1900) or so

Reply:
I am considering removing this altogether, as klemke also has some concerns.